src/sentry/deletions/defaults/group.py

```diff
@@ -159,6 +164,53 @@ def delete_events_from_snuba(self) -> None:
             eventstream.backend.end_delete_groups(eventstream_state)
 
 
+class IssuePlatformEventsDeletionTask(EventsBaseDeletionTask):
+    """
+    This class helps delete Issue Platform events which use the new Clickhouse light deletes.
+    """
+
+    dataset = Dataset.IssuePlatform
+
+    def chunk(self) -> bool:
+        """This method is called to delete chunks of data. It returns a boolean to say
+        if the deletion has completed and if it needs to be called again."""
+        events = self.get_unfetched_events()
+        if events:
+            self.delete_events_from_nodestore(events)
+            # This value will be used in the next call to chunk
+            self.last_event = events[-1]
+            # As long as it returns True the task will keep iterating
+            return True
+        else:
+            # Now that all events have been deleted from the eventstore, we can delete the events from snuba
+            self.delete_events_from_snuba()
+            return False
+
+    def delete_events_from_nodestore(self, events: Sequence[Event]) -> None:
+        # We delete by the occurrence_id instead of the event_id
+        node_ids = [
+            Event.generate_node_id(event.project_id, event._snuba_data["occurrence_id"])
+            for event in events
+        ]
+        nodestore.backend.delete_multi(node_ids)
```

Comment:
This deletes the occurrence from nodestore. Does it delete the event associated with the occurrence too?

Note that we should only delete the event if it was created during occurrence ingestion. If we passed an event id, then the event should remain.

Reply:
We only delete the occurrence event from the Issue Platform dataset (not the Events one). The error event from the Events dataset is left untouched.

See the test logic below:
```python
        # The original event and group still exist
        assert Group.objects.filter(id=event.group_id).exists()
        event_node_id = Event.generate_node_id(event.project_id, event.event_id)
        assert nodestore.backend.get(event_node_id)

        # The Issue Platform group and occurrence are deleted
        assert issue_platform_group.issue_type == FeedbackGroup
        assert not Group.objects.filter(id=issue_platform_group.id).exists()
        occurrence_node_id = Event.generate_node_id(occurrence.project_id, occurrence.id)
        assert not nodestore.backend.get(occurrence_node_id)
        # Assert that occurrence is gone
        assert self.select_issue_platform_events(self.project.id) == []
```

Reply:
Under what circumstances does the event get created during occurrence ingestion? I don't think we spend time finding the code for that.

Last time we spoke we agreed to keep it simple and let the event expire with the TTL.

Reply:
> Under what circumstances does the event get created during occurrence ingestion? I don't think we spend time finding the code for that.

It is hard to tell the difference between the two post ingestion - we don't differentiate since the occurrence just has an event associated with it. The split happens here: https://github.com/getsentry/sentry/blob/a86b9b672709bc9c4558cffb2c825965b8cee0d1/src/sentry/issues/occurrence_consumer.py#L324-L339

> Last time we spoke we agreed to keep it simple and let the event expire with the TTL.

I don't specifically remember that, but since these events can't be accessed in any reasonable way maybe that's fine, since there's nothing linking to them. I think we should probably mention that explicitly in a comment here if that's the case so that we're aware.

Reply:
> We only delete the occurrence event from the Issue Platform dataset (not the Events one). The error event from the Events dataset is left untouched.
> 
> See the test logic below:
> 
> ```python
>         # The original event and group still exist
>         assert Group.objects.filter(id=event.group_id).exists()
>         event_node_id = Event.generate_node_id(event.project_id, event.event_id)
>         assert nodestore.backend.get(event_node_id)
>         assert self.select_error_events(self.project.id) == error_expected
>         # The Issue Platform group and occurrence are deleted
>         assert issue_platform_group.issue_type == FeedbackGroup
>         assert not Group.objects.filter(id=issue_platform_group.id).exists()
>         occurrence_node_id = Event.generate_node_id(occurrence.project_id, occurrence.id)
>         assert not nodestore.backend.get(occurrence_node_id)
>         # Assert that occurrence is gone
>         assert self.select_issue_platform_events(self.project.id) is None
> ```

I'm not talking about the events dataset, just the event in Nodestore.

Reply:
@wedamija We check in the test that it is not deleted from either

Reply:
Thanks, @wedamija for all your help!

I have added a comment to try to encompass what you explained.

<img width="909" alt="image" src="https://github.com/user-attachments/assets/a415c98c-66c0-4198-bd17-0220938658a7">